lib/worker.js

```diff
@@ -69,8 +78,23 @@ class Worker extends BaseService {
             this._metrics = undefined;
         }
         this._heapwatchHandle.close();
-        if (Array.isArray(this.serviceReturns)) {
-            return P.each(this.serviceReturns, (serviceRet) => {
+
+        // Remove signal handlers
+        if (this._shutdownHandler) {
+            process.removeListener('SIGTERM', this._shutdownHandler);
+        }
+        if (this._dumpHeapHandler) {
+            process.removeListener('SIGUSR2', this._dumpHeapHandler);
+        }
+        if (this._serviceStatusHandler) {
+            process.removeListener('service_status', this._serviceStatusHandler);
+        }
+        if (this._messageHandler) {
+            process.removeListener('message', this._messageHandler);
+        }
```

Comment:
We need to de-initialize the fields too and set them to undefined.

Reply:
Done